### PR TITLE
fix(shortcode): give_totals should output the correct donation total #3291

### LINKED ISSUE #3291: fix(shortcode): give_totals should output the correct donation total
## User Story
As a user when I would use the `give_totals` shortcode, I accept it to show the correct total which should be the sum of all the donations for the donation forms included.

For example: If you have a donation form in English which is translated in French and have donated 11 USD in the form in English and 22 USD in the form in French.

Now if you use the `give_totals` shortcode and include the IDs of this two form.

In the page which is using the shortcode, It will display the total as 11 USD for English and 22 USD for French instead of 33USD.

## Current Behavior
It should display the correct donation total.

## Steps to Reproduce
I have created a doc which explains in details how to replicate this issue 
https://docs.google.com/document/d/1-xeTxHn5K9yeYbRniV7T5BT6mYq_v3WG9BHehivBa8Y/edit

## Related
Link to the HS ticket: https://secure.helpscout.net/conversation/574804454/17963?folderId=1457790

## Environment
<details>
 ### WordPress Environment ###

Home URL: http://give-wpml.local
Site URL: http://give-wpml.local
WP Version: 4.9.6
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
Permalink Structure: /%postname%/
Show on Front: posts
Table Prefix Length: wp_
Table Prefix Length: 3
Table Prefix Status: Acceptable
Admin AJAX: Accessible
Registered Post Statuses: publish, future, draft, pending, private, trash, auto-draft, inherit, request-pending, request-confirmed, request-failed, request-completed, refunded, failed, revoked, cancelled, abandoned, processing, preapproval

### Server Environment ###

Hosting Provider: DBH: localhost, SRV: give-wpml.local
TLS Connection: Warning: file_get_contents(): SSL operation failed with code 1. OpenSSL Error messages:
error:14090086:SSL routines:ssl3_get_server_certificate:certificate verify failed in /app/public/wp-content/plugins/give/includes/admin/tools/views/html-admin-page-system-info.php on line 188

Warning: file_get_contents(): Failed to enable crypto in /app/public/wp-content/plugins/give/includes/admin/tools/views/html-admin-page-system-info.php on line 188

Warning: file_get_contents(https://www.howsmyssl.com/a/check): failed to open stream: operation failed in /app/public/wp-content/plugins/give/includes/admin/tools/views/html-admin-page-system-info.php on line 188
TLS Connection: 
Server Info: nginx/1.10.1
PHP Version: 7.1.7
PHP Post Max Size: 1,000 MB
PHP Time Limit: 1200
PHP Max Input Vars: 4000
PHP Max Upload Size: 300 MB
cURL Version: ❌ 7.38.0, OpenSSL/1.0.2l - We recommend a minimum cURL version of 7.40.
SUHOSIN Installed: –
MySQL Version: 5.6.37
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
gzip: ✔
GD Graphics Library: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Give Configuration ###

Give Version: 2.1.3
Give Cache: Enabled
Database Updates: All DB Updates Completed.
Give Cache: Enabled
Give Cache: ✔New Donation✔Donation Receipt✔New Offline Donation✔Offline Donation Instructions✔New User Registration✔User Registration Information✔Email access
Upgraded From: –
Test Mode: Enabled
Currency Code: USD
Currency Position: Before
Decimal Separator: .
Thousands Separator: ,
Success Page: http://give-wpml.local/donation-confirmation/
Failure Page: http://give-wpml.local/donation-failed/
Donation History Page: http://give-wpml.local/donation-history/
Give Forms Slug: /donations/
Enabled Payment Gateways: Test Donation, Offline Donation
Default Payment Gateway: Test Donation
PayPal IPN Verification: Enabled
PayPal IPN Notifications: N/A
Donor Email Access: Enabled

### Session Configuration ###

Give Use Sessions: Enabled
Session: Disabled

### Other Active Plugins ###

WPML Multilingual CMS: by OnTheGoSystems – 3.7.1

### Theme ###

Name: Twenty Seventeen
Version: 1.6
Author URL: https://wordpress.org/
Child Theme: No – If you're modifying Give on a parent theme you didn't build personally, then we recommend using a child theme. See: How to Create a Child Theme

</details>

### PR DESCRIPTION
**Pull request by raftaar1191**
## Description
PR to fix #3291 

## How Has This Been Tested?
Have tested by adding two donation form of the same type but the language is diff and runs on the same page with two diff language s

## Screenshots (jpeg or gifs if applicable):
Video Link: https://screencast-o-matic.com/watch/cF1V1QbLWa

image

image

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: includes/actions.php [hunk 1]
```diff
@@ -389,3 +389,82 @@ function __give_verify_addon_dependency_before_update( $error, $hook_extra ) {
...
 add_filter( 'upgrader_pre_install', '__give_verify_addon_dependency_before_update', 10, 2 );
+
+/**
+ * Function to remove WPML post where filter in goal total amount shortcode.
+ *
+ * @since 2.1.4
+ */
+function give_remove_wpml_posts_where_filter() {
+	global $wpml_query_filter;
+	remove_filter( 'posts_where', array( $wpml_query_filter, 'posts_where_filter' ), 10, 2 );
```
Comment (ravinderk) [inline includes/actions.php]: @raftaar1191 remove extra priority param

File: includes/actions.php [hunk 2]
```diff
@@ -389,3 +389,82 @@ function __give_verify_addon_dependency_before_update( $error, $hook_extra ) {
...
+}
+
+/**
+ * Function to add WPML post where filter in goal total amount shortcode.
+ *
+ * @since 2.1.4
+ */
+function give_add_wpml_posts_where_filter() {
+	global $wpml_query_filter;
+	add_filter( 'posts_where', array( $wpml_query_filter, 'posts_where_filter' ), 10, 2 );
```
Comment (ravinderk) [inline includes/actions.php]: @raftaar1191 remove add filter 

```
remove_filter( 'wpml_pre_parse_query', 'give_wpml_pre_parse_query', 10 );
remove_filter( 'wpml_post_parse_query', 'give_wpml_post_parse_query', 10 );
```

File: includes/actions.php [hunk 3]
```diff
@@ -389,3 +389,82 @@ function __give_verify_addon_dependency_before_update( $error, $hook_extra ) {
...
+/**
+ * Alter the WP query argument for getting over write by WPML.
+ *
+ * @since 2.1.4
+ *
+ * @param WP_Query $q
+ *
+ * @return WP_Query
+ */
+function give_wpml_post_parse_query( $q ) {
```
Comment (ravinderk) [inline includes/actions.php]: @raftaar1191 use `__` add fn prefix because they are internal fn

File: includes/shortcodes.php [hunk 4]
```diff
@@ -596,6 +596,15 @@ function give_totals_shortcode( $atts ) {
...
 	$total_goal = give_maybe_sanitize_amount( $atts['total_goal'] );
 
+	/**
+	 * Give Action fire before the shortcode is rendering is started.
+	 *
+	 * @since 2.1.4
+	 *
+	 * @param array $atts shortcode attribute.
+	 */
+	do_action( 'give_totals_goal_shortcode_start', $atts );
```
Comment (ravinderk) [inline includes/shortcodes.php]: @raftaar1191  rename it to `give_totals_goal_shortcode_before_render`

File: includes/shortcodes.php [hunk 5]
```diff
@@ -699,6 +708,16 @@ function give_totals_shortcode( $atts ) {
...
 
+	/**
+	 * Give Action fire after the total goal shortcode rendering is end.
+	 *
+	 * @since 2.1.4
+	 *
+	 * @param array $atts shortcode attribute.
+	 * @param HTML $give_totals_output shortcode output.
+	 */
+	do_action( 'give_totals_goal_shortcode_end', $atts, $give_totals_output );
```
Comment (ravinderk) [inline includes/shortcodes.php]: @raftaar1191  rename it to `give_totals_goal_shortcode_after_render`

File: includes/actions.php (hunk 1 above)
Comment (raftaar1191) [reply includes/actions.php]: Removed @ravinderk

File: includes/actions.php (hunk 2 above)
Comment (raftaar1191) [reply includes/actions.php]: Added @ravinderk

File: includes/shortcodes.php (hunk 5 above)
Comment (raftaar1191) [reply includes/shortcodes.php]: Changed @ravinderk

File: includes/shortcodes.php (hunk 4 above)
Comment (raftaar1191) [reply includes/shortcodes.php]: Changed @ravinderk

File: includes/actions.php (hunk 3 above)
Comment (raftaar1191) [reply includes/actions.php]: Added @ravinderk

**Comment by raftaar1191** [comment]
@ravinderk All the suggestions that are being suggested by you are being Done 


I try to regenerate the notices as well but I don't think this is related to to the current PR

**Comment by ravinderk** [comment]
@raftaar1191  use `suppress_filters` filters as we discuss overcall and remove existing code.

**Comment by raftaar1191** [comment]
@ravinderk PR updated as suggested by you but we need to add still add some filter because when we pass donation id in the Total Goal shortcode it is not working 


Please check the video to know about the issues and why I need to add that extra filter 

Video Link: https://screencast-o-matic.com/watch/cF1VIqbNow

**Pull request merged**